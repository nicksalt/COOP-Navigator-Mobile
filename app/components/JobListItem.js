import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import Icon from "react-native-vector-icons/Ionicons";

export default class JobListItem extends React.Component {
  render() {
    return (
      <TouchableOpacity onPress={this._itemOnClick}>
        <View style={styles.parentView}>
          <View>
            <Text style={styles.heading}>
              {this.props.job.organization}
              {"\n"}
            </Text>
            <Text style={styles.subHeading}>
              {this.props.job.englishTitle}
              {"\n"}
            </Text>
            <Text style={styles.jobid}>
              {"Job ID: " + this.props.job.jobId}
            </Text>
          </View>
          <View style={styles.chevronContainer}>
            <Icon name="ios-arrow-forward" style={styles.chevron} />
          </View>
        </View>
      </TouchableOpacity>
    );
  }

  _itemOnClick = () => {
    this.props.fetchSingleJobDispatch(this.props.job.jobId);
    this.props.navigation.navigate("JobSummary");
  };
}

JobListItem.propTypes = {
  job: PropTypes.shape({
    jobId: PropTypes.number,
    englishTitle: PropTypes.string,
    frenchTitle: PropTypes.string,
    organization: PropTypes.string
  })
};

const styles = StyleSheet.create({
  parentView: {
    padding: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between"
  },
  header: {},
  heading: {
    color: "#a9343a",
    fontSize: 18
  },
  subHeading: {
    fontSize: 18,
    flexDirection: "row"
  },
  jobid: {
    fontSize: 12,
    color: "gray"
  },
  chevronContainer: {
    flexDirection: "column",
    justifyContent: "center"
  },
  chevron: {
    fontSize: 25,
    paddingRight: 8,
    color: "#868686",
    justifyContent: "flex-end"
  }
});
