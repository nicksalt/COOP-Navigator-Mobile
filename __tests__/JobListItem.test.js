import React from "react";
import JobListItem from "../app/components/JobListItem";
import renderer from "react-test-renderer";

test("JobListItem snapshot renders correctly", () => {
  let props = {
    job: {
      jobId: 100,
      englishTitle: "Job In English",
      frenchTitle: "Job En Francais",
      organization: "A Fancy Org"
    }
  };
  const tree = renderer.create(<JobListItem {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
