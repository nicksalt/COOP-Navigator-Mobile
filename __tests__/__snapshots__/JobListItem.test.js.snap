// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobListItem snapshot renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#a9343a",
            "fontSize": 18,
          }
        }
      >
        A Fancy Org
        

      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flexDirection": "row",
            "fontSize": 18,
          }
        }
      >
        Job In English
        

      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "gray",
            "fontSize": 12,
          }
        }
      >
        Job ID: 100
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</View>
`;
